src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -236,6 +236,29 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let result = Scalar::from_uint(truncated_bits, layout.size);
                 self.write_scalar(result, dest)?;
             }
+
+            "ptr_offset_from" => {
+                let a = self.read_immediate(args[0])?.to_scalar()?.to_ptr()?;
+                let b = self.read_immediate(args[1])?.to_scalar()?.to_ptr()?;
```

Comment:
Ah, I just realized this is wrong... we need to support integers here as well.

This caused a regression in https://github.com/RalfJung/miri-test-libstd.

Reply:
https://github.com/rust-lang/rust/pull/66083 fixes that.